server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -2627,6 +2641,12 @@ public Pair<List<? extends GuestOSHypervisor>, Integer> listGuestOSMappingByCrit
             sc.addAnd("hypervisorVersion", SearchCriteria.Op.EQ, hypervisorVersion);
         }
 
+        if (osDisplayName != null) {
+            List<GuestOSVO> guestOSVOS = _guestOSDao.listLikeDisplayName(osDisplayName);
+            List<Long> guestOSids = guestOSVOS.stream().map(mo -> mo.getId()).collect(Collectors.toList());
+            sc.addAnd("guestOsId", SearchCriteria.Op.IN, guestOSids.toArray());
```

Comment:
@harikrishna-patnala 
you may need to check if guestOSids is empty here.
I have seen some mysql errors like "name in ( )" in the past.